emmet-core/emmet/core/openmm/tasks.py

```diff
@@ -0,0 +1,217 @@
+"""Schemas for OpenMM tasks."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Union
+
+import pandas as pd  # type: ignore[import-untyped]
+from emmet.core.vasp.task_valid import TaskState  # type: ignore[import-untyped]
+from pydantic import BaseModel, Field
+
+from atomate2.classical_md.schemas import ClassicalMDTaskDocument
+
+
+class CalculationInput(BaseModel, extra="allow"):  # type: ignore[call-arg]
+    """OpenMM input settings for a job, these are the attributes of the OpenMMMaker."""
+
+    n_steps: Optional[int] = Field(
+        None, description="The number of simulation steps to run."
+    )
+
+    step_size: Optional[float] = Field(
+        None, description="The size of each simulation step (picoseconds)."
+    )
+
+    temperature: Optional[float] = Field(
+        None, description="The simulation temperature (kelvin)."
+    )
+
+    friction_coefficient: Optional[float] = Field(
+        None,
+        description=(
+            "The friction coefficient for the integrator " "(inverse picoseconds)."
+        ),
+    )
+
+    platform_name: Optional[str] = Field(
+        None,
+        description=(
+            "The name of the OpenMM platform to use, passed to "
+            "Interchange.to_openmm_simulation."
+        ),
+    )
+
+    platform_properties: Optional[dict] = Field(
+        None,
+        description=(
+            "Properties for the OpenMM platform, passed to "
+            "Interchange.to_openmm_simulation."
+        ),
+    )
+
+    state_interval: Optional[int] = Field(
+        None,
+        description=(
+            "The interval for saving simulation state. For no state, set to 0."
+        ),
+    )
+
+    state_file_name: Optional[str] = Field(
+        None, description="The name of the state file to save."
+    )
+
+    traj_interval: Optional[int] = Field(
+        None,
+        description=(
+            "The interval for saving trajectory frames. For no trajectory, set to 0."
+        ),
```

Comment:
Same comment about units. If there are no units and this is the number of timesteps, that would also be good to specify.